pkg/features/builder.go

```diff
@@ -34,7 +34,7 @@ func New(name string) *FeatureBuilder {
 
 // WithLabel adds a test label key/value pair
 func (b *FeatureBuilder) WithLabel(key, value string) *FeatureBuilder {
-	b.feat.labels[key] = value
+	b.feat.labels[key] = append(b.feat.labels[key], value)
```

Comment:
Since we process the argument keys on label related attribute as lower case, would it make sense to turn the key here as well into lowercase and store it in a uniform way ?